# Comparison of MLE and GC results from SDS x CS82

After matching the SDStripe82 catalog with CS82 using both xmatching algorithms: `mle` and `gc`, we will now check how their results change -- if at all.

The catalogs are stored in the following files:
* gc: `swift_deepsky/xcat_sds_cs82_gc_7arcsec.csv`
* mle: `swift_deepsky/xcat_sds_cs82_mle_MAG_AUTO.csv`

In [1]:
gc_catalog = 'swift_deepsky/xcat_sds_cs82_gc_7arcsec.csv'
mle_catalog = 'swift_deepsky/xcat_sds_cs82_mle_MAG_AUTO.csv'

In [4]:
import pandas
pandas.set_option('display.max_rows',10)

df_gc = pandas.read_csv(gc_catalog, index_col=0, header=[0,1])
df_mle = pandas.read_csv(mle_catalog, index_col=0, header=[0,1])

In [5]:
df_gc

A                           B                          \
             RA       DEC OBJID ALPHA_J2000 DELTA_J2000       OBJID   
0     14.102000 -1.277310     0         NaN         NaN         NaN   
1     14.079733 -1.249499     1         NaN         NaN         NaN   
2     14.095850 -1.227643     2         NaN         NaN         NaN   
3     14.068408 -1.234838     3         NaN         NaN         NaN   
4     14.070508 -1.221223     4         NaN         NaN         NaN   
...         ...       ...   ...         ...         ...         ...   
2759  17.460071  0.393572  6634   17.460878    0.392982   9439689.0   
2760  39.237083  0.726716  6732   39.236618    0.726102  14787476.0   
2761  39.230850  0.613294  6733   39.232101    0.613970  14775413.0   
2762  39.304346  0.671991  6734   39.305242    0.672597  14781889.0   
2763  39.137842  0.527952  6736   39.138360    0.528292  14764746.0   

             AB                           \
     separation               duplicates   
0           NaN                      NaN   
1           NaN                      NaN   
2           NaN                      NaN   
3           NaN                      NaN   
4           NaN                      NaN   
...         ...                      ...   
2759   3.598603  9439942;9439963;9440078   
2760   2.772516        14787240;14787374   
2761   5.117992                 14775246   
2762   3.893721        14781782;14781801   
2763   2.232993        14764938;14765002   

                                                         
                                              distances  
0                                                   NaN  
1                                                   NaN  
2                                                   NaN  
3                                                   NaN  
4                                                   NaN  
...                                                 ...  
2759  5.14770756999403;5.3795387873798;6.57092331381...  
2760               4.210032238557784;4.4539534358460795  
2761                                 6.5627534729293835  
2762                3.960413624509272;4.705623829117702  
2763                6.4949025059168735;2.78894605390959  

[2764 rows x 9 columns]

In [6]:
df_mle

A                                 B                          \
                RA       DEC OBJID_A pos_err OBJID_B ALPHA_J2000 DELTA_J2000   
OBJID_A                                                                        
0        14.102000 -1.277310     0.0     5.0     NaN         NaN         NaN   
1        14.079733 -1.249499     1.0     5.0     NaN         NaN         NaN   
2        14.095850 -1.227643     2.0     5.0     NaN         NaN         NaN   
3        14.068408 -1.234838     3.0     5.0     NaN         NaN         NaN   
4        14.070508 -1.221223     4.0     5.0     NaN         NaN         NaN   
...            ...       ...     ...     ...     ...         ...         ...   
6869     57.205183 -1.076907  6869.0     5.0     NaN         NaN         NaN   
6870     57.102754 -0.952921  6870.0     5.0     NaN         NaN         NaN   
6875     57.145083 -1.066406  6875.0     5.0     NaN         NaN         NaN   
6876     57.190829 -1.100567  6876.0     5.0     NaN         NaN         NaN   
6877     57.176250 -1.095605  6877.0     5.0     NaN         NaN         NaN   

                 AB_MAG_AUTO                                            
        MAG_AUTO Reliability  LR duplicates duplicates_LR duplicates_R  
OBJID_A                                                                 
0            NaN         NaN NaN        NaN           NaN          NaN  
1            NaN         NaN NaN        NaN           NaN          NaN  
2            NaN         NaN NaN        NaN           NaN          NaN  
3            NaN         NaN NaN        NaN           NaN          NaN  
4            NaN         NaN NaN        NaN           NaN          NaN  
...          ...         ...  ..        ...           ...          ...  
6869         NaN         NaN NaN        NaN           NaN          NaN  
6870         NaN         NaN NaN        NaN           NaN          NaN  
6875         NaN         NaN NaN        NaN           NaN          NaN  
6876         NaN         NaN NaN        NaN           NaN          NaN  
6877         NaN         NaN NaN        NaN           NaN          NaN  

[2764 rows x 13 columns]

## Let's check how many matching-pairs match between the catalogs

* This verification should gives us a measure on how much those two methods actually differ, and as such, an estimate of the xmatching error. 
* And it should point us to interesting sources or regions, where the nearest neighbor is *not* the "right" match.

In [7]:
objids_gc = df_gc[[('A','OBJID'),('B','OBJID')]]
objids_gc = objids_gc.dropna()
objids_gc = objids_gc.astype(int)
objids_gc.columns = ['OBJID_A','OBJID_B']
objids_gc.set_index('OBJID_A', inplace=True)

objids_gc.head()

OBJID_B
OBJID_A          
14        6203717
45       10331489
46       10331779
47       10336933
48       10349649

In [8]:
objids_mle = df_mle[[('A','OBJID_A'),('B','OBJID_B')]]
objids_mle = objids_mle.dropna()
objids_mle = objids_mle.astype(int)
objids_mle.columns = ['OBJID_A','OBJID_B']
objids_mle.set_index('OBJID_A', inplace=True)

objids_mle.head()

OBJID_B
OBJID_A          
14        6203632
45       10331489
46       10331970
47       10336874
48       10349673

In [9]:
all(objids_mle.index.sort_values() == objids_gc.index.sort_values())

True

In [10]:
dfx = objids_gc.join(objids_mle, how='outer', lsuffix='_gc', rsuffix='_mle')
dfx.head()

OBJID_B_gc  OBJID_B_mle
OBJID_A                         
14          6203717      6203632
45         10331489     10331489
46         10331779     10331970
47         10336933     10336874
48         10349649     10349673

In [11]:
df_diff = dfx['OBJID_B_gc'] == dfx['OBJID_B_mle']
df_diff.head()

OBJID_A
14    False
45     True
46    False
47    False
48    False
dtype: bool

In [12]:
equal_matches = df_diff.sum() / len(df_diff)

print('{:.1f}% of matches coincide between "gc" and "mle"'.format(equal_matches * 100))

70.4% of matches coincide between "gc" and "mle"


**70.4% of matches coincide between "gc" and "mle"**

## Who are the non-matching objects from 'mle'?

Let's check some properties of those ~30% objects that matched using the likelihood estimator.

In [13]:
_idx = df_diff.loc[~df_diff].index
df_nonmatching = df_mle.set_index(('A','OBJID_A')).loc[_idx]
df_nonmatching.head()

A                             B                          \
                RA       DEC pos_err     OBJID_B ALPHA_J2000 DELTA_J2000   
OBJID_A                                                                    
14       -3.756921 -0.026718     5.0   6203632.0   -3.755398   -0.027458   
46       24.657625 -0.668684     5.0  10331970.0   24.656740   -0.668031   
47       24.819667 -0.621067     5.0  10336874.0   24.821171   -0.621196   
48       24.696104 -0.495326     5.0  10349673.0   24.695655   -0.495929   
52       24.863604 -0.518196     5.0  10347523.0   24.863112   -0.518832   

                   AB_MAG_AUTO                                        \
          MAG_AUTO Reliability        LR                  duplicates   
OBJID_A                                                                
14       19.368633    0.764507  0.019682             6203717;6203774   
46       21.149553    0.446835  0.012284           10331779;10331716   
47       21.617054    0.399341  0.007689  10336933;10336804;10336854   
48       20.810715    0.598017  0.016153  10349649;10349580;10349671   
52       22.349245    0.565719  0.007371                    10347561   

                                                            \
                                             duplicates_LR   
OBJID_A                                                      
14              0.0024608894214439452;0.003601914698632182   
46                0.00490188752174791;0.010305705467634313   
47       0.0025365640202913682;0.0037324721520687115;0....   
48       0.00479203354230448;0.0028684722636638463;0.00...   
52                                    0.005658521449655007   

                                                            
                                              duplicates_R  
OBJID_A                                                     
14                     0.09558660098397;0.1399066451650775  
46                  0.17830270990683272;0.3748627858612379  
47       0.131745698030017;0.19385954587316417;0.275053...  
48       0.177410883069921;0.10619671019945533;0.118375...  
52                                      0.4342814700655355

In [14]:
29.59 < 100*len(df_nonmatching)/len(df_diff) < 29.60

True

In [15]:
from bokeh.io import output_notebook

output_notebook()

Loading BokehJS ...

In [20]:
def plot_distros(data1, data2, legend1, legend2):
    from bokeh.plotting import figure

    p = figure(plot_width=600, plot_height=400)

    bins = (data1.min(), data1.max(), 50)

    p = histogram(p, data1.dropna(), color='yellow', bins=bins, legend=legend1)
    p = histogram(p, data2.dropna(), color='blue', bins=bins, legend=legend2)
    p.legend.location = "top_left"
    
    return p

In [21]:
# HISTOGRAM

def histogram(fig, data, color, bins, legend):
    import numpy as np
    
    bins = np.linspace(*bins)
    
    hist, edges = np.histogram(data, density=True, bins=bins)
    
    fig.quad(top=hist, bottom=0, left=edges[:-1], right=edges[1:], 
             fill_color=color, line_color="#033649", fill_alpha=0.5,
             legend=legend)
    
    return fig

## .. a bigger sample from CS82

In [22]:
df_cs82_reduced = pandas.read_csv('cs82_photoauto_reduced_for_sds.csv', index_col=0)
df_cs82_reduced.head()

OBJID_B  ALPHA_J2000  DELTA_J2000   MAG_AUTO
0   6203717    -3.756927    -0.025511  25.151508
1  10331489    24.693307    -0.672908  21.615351
2  10331779    24.656872    -0.669435  23.783859
3  10336933    24.819401    -0.620192  25.077038
4  10349649    24.696638    -0.495811  24.011000

#### Plot: Background sample: 'MAG_AUTO' distribution of background sources (yellow) and counterparts (blue)

In [23]:
from bokeh.io import show

_col = ('B','MAG_AUTO')

data1 = df_cs82_reduced[_col[1]].dropna()
data2 = df_mle[_col].dropna()
legend1 = 'background ({:d})'.format(len(data1))
legend2 = 'mle ({:d})'.format(len(data2))

p = plot_distros(data1, data2, legend1, legend2)
p.title.text = 'CS82 magnitude distribution of matched-vs-all sources'
p.title.align = 'center'
p.xaxis.axis_label = 'Magnitude (MAG_AUTO)'
p.yaxis.axis_label = '#'

show(p)

#### Plot: Background sample: 'MAG_AUTO' distribution of background sources (yellow) and the gc-gc (blue) sub-sample

In [24]:
_idx = df_diff.loc[df_diff].index
df_matching = df_mle.set_index(('A','OBJID_A')).loc[_idx]
df_matching.head()

from bokeh.io import show

_col = ('B','MAG_AUTO')

data1 = df_cs82_reduced[_col[1]].dropna()
data2 = df_matching[_col].dropna()
legend1 = 'background ({:d})'.format(len(data1))
legend2 = 'gc & gc ({:d})'.format(len(data2))

p = plot_distros(data1, data2, legend1, legend2)

p.title.text = 'CS82 magnitude distribution of matched-vs-all sources'
p.title.align = 'center'
p.xaxis.axis_label = 'Magnitude (MAG_AUTO)'
p.yaxis.axis_label = '#'

show(p)

#### Plot: Background sample: 'MAG_AUTO' distribution of background sources (yellow) and the mle-gc (blue) sub-sample

In [25]:
from bokeh.io import show

_col = ('B','MAG_AUTO')

data1 = df_cs82_reduced[_col[1]].dropna()
data2 = df_nonmatching[_col].dropna()
legend1 = 'background ({:d})'.format(len(data1))
legend2 = 'mle-gc ({:d})'.format(len(data2))

p = plot_distros(data1, data2, legend1, legend2)

p.title.text = 'CS82 magnitude distribution of matched-vs-all sources'
p.title.align = 'center'
p.xaxis.axis_label = 'Magnitude (MAG_AUTO)'
p.yaxis.axis_label = '#'

show(p)

#### Plot: Cross-matched sample: 'MAG_AUTO' distribution of all sources (yellow) and the mle-only (blue) sub-sample

In [26]:
from bokeh.io import show

_col = ('B','MAG_AUTO')

data1 = df_mle[_col]
data2 = df_nonmatching[_col]
legend1 = 'all matched'
legend2 = 'mle only (mle-gc)'
p = plot_distros(data1, data2, legend1, legend2)

show(p)

## The GC catalog

In [27]:
df_cs82_gc = df_cs82_reduced.set_index('OBJID_B').loc[objids_gc['OBJID_B']]
df_cs82_gc.head()

ALPHA_J2000  DELTA_J2000   MAG_AUTO
OBJID_B                                      
6203717     -3.756927    -0.025511  25.151508
10331489    24.693307    -0.672908  21.615351
10331779    24.656872    -0.669435  23.783859
10336933    24.819401    -0.620192  25.077038
10349649    24.696638    -0.495811  24.011000

## Separation (GC) between pairs

In [28]:
def plot_distro(data):
    from bokeh.plotting import figure

    p = figure(plot_width=600, plot_height=400)

    bins = (data.min(), data.max(), 50)

    p = histogram(p, data.dropna(), color='blue', bins=bins, legend=None)
    
    return p

In [29]:
p = plot_distro(df_gc[('AB','separation')])
p.title.text = 'Distance separation between matching sources in SDS-x-CS82'
p.title.align = 'center'
p.xaxis.axis_label = 'Separation (arcsec)'
p.yaxis.axis_label = '#'

In [30]:
show(p)

In [31]:
df_gc[('AB','separation')].describe()

count    1105.000000
mean        2.901299
std         1.569761
min         0.052965
25%         1.700351
50%         2.643995
75%         3.864710
max         6.971090
Name: (AB, separation), dtype: float64

In [32]:
from bokeh.io import show

_col = ('B','MAG_AUTO')

data1 = df_cs82_reduced[_col[1]].dropna()
data2 = df_cs82_gc[_col[1]].dropna()
legend1 = 'background ({:d})'.format(len(data1))
legend2 = 'gc ({:d})'.format(len(data2))

p = plot_distros(data1, data2, legend1, legend2)

p.title.text = 'CS82 magnitude distribution of matched-vs-all sources'
p.title.align = 'center'
p.xaxis.axis_label = 'Magnitude (MAG_AUTO)'
p.yaxis.axis_label = '#'

show(p)

# Let's check the x-ray flux behaviour

In [33]:
sds_file = 'swift_deepsky/table_flux_detections_stripe82_unique.csv'

In [34]:
import pandas

df_sds = pandas.read_csv(sds_file, sep=';')
df_sds.head()

OBJID         RA       DEC            NH  ENERGY_SLOPE ENERGY_SLOPE_ERROR  \
0      0  14.102000 -1.277310  3.540000e+20           0.8          -999/-999   
1      1  14.079733 -1.249499  3.520000e+20           0.8          -999/-999   
2      2  14.095850 -1.227643  3.510000e+20           0.8          -999/-999   
3      3  14.068408 -1.234838  3.510000e+20           0.8          -999/-999   
4      4  14.070508 -1.221223  3.500000e+20           0.8          -999/-999   

   EXPOSURE_TIME  nufnu_3keV(erg.s-1.cm-2)  nufnu_error_3keV(erg.s-1.cm-2)  \
0         4572.8              7.969610e-14                    2.117700e-14   
1         4684.2              6.277090e-14                    1.975260e-14   
2         4649.6              5.501970e-14                    1.974560e-14   
3         4642.0              8.173270e-14                    2.256640e-14   
4         4587.3              7.365840e-14                    2.115000e-14   

   nufnu_0.5keV(erg.s-1.cm-2)  nufnu_error_0.5keV(erg.s-1.cm-2)  \
0                2.247330e-14                      1.552740e-14   
1                1.277270e-14                      1.252210e-14   
2                2.238590e-14                      1.751890e-14   
3                1.151060e-14                      1.187720e-14   
4                5.616880e-14                      2.418840e-14   

   upper_limit_0.5keV(erg.s-1.cm-2)  nufnu_1.5keV(erg.s-1.cm-2)  \
0                            -999.0                1.088260e-13   
1                            -999.0                7.075490e-14   
2                            -999.0                4.655630e-14   
3                            -999.0                9.568910e-14   
4                            -999.0                4.673000e-14   

   nufnu_error_1.5keV(erg.s-1.cm-2)  upper_limit_1.5keV(erg.s-1.cm-2)  \
0                      4.027270e-14                            -999.0   
1                      3.430380e-14                            -999.0   
2                      3.005940e-14                            -999.0   
3                      3.936350e-14                            -999.0   
4                      2.649940e-14                            -999.0   

   nufnu_4.5keV(erg.s-1.cm-2)  nufnu_error_4.5keV(erg.s-1.cm-2)  \
0                9.370070e-14                      4.601960e-14   
1                1.066430e-13                      5.170340e-14   
2                9.353370e-14                      5.227960e-14   
3                1.442380e-13                      5.933510e-14   
4                9.390950e-14                      4.601620e-14   

   upper_limit_4.5keV(erg.s-1.cm-2)  
0                            -999.0  
1                            -999.0  
2                            -999.0  
3                            -999.0  
4                            -999.0

In [35]:
df_sds.set_index('OBJID', inplace=True)
df_sds.head()

RA       DEC            NH  ENERGY_SLOPE ENERGY_SLOPE_ERROR  \
OBJID                                                                       
0      14.102000 -1.277310  3.540000e+20           0.8          -999/-999   
1      14.079733 -1.249499  3.520000e+20           0.8          -999/-999   
2      14.095850 -1.227643  3.510000e+20           0.8          -999/-999   
3      14.068408 -1.234838  3.510000e+20           0.8          -999/-999   
4      14.070508 -1.221223  3.500000e+20           0.8          -999/-999   

       EXPOSURE_TIME  nufnu_3keV(erg.s-1.cm-2)  \
OBJID                                            
0             4572.8              7.969610e-14   
1             4684.2              6.277090e-14   
2             4649.6              5.501970e-14   
3             4642.0              8.173270e-14   
4             4587.3              7.365840e-14   

       nufnu_error_3keV(erg.s-1.cm-2)  nufnu_0.5keV(erg.s-1.cm-2)  \
OBJID                                                               
0                        2.117700e-14                2.247330e-14   
1                        1.975260e-14                1.277270e-14   
2                        1.974560e-14                2.238590e-14   
3                        2.256640e-14                1.151060e-14   
4                        2.115000e-14                5.616880e-14   

       nufnu_error_0.5keV(erg.s-1.cm-2)  upper_limit_0.5keV(erg.s-1.cm-2)  \
OBJID                                                                       
0                          1.552740e-14                            -999.0   
1                          1.252210e-14                            -999.0   
2                          1.751890e-14                            -999.0   
3                          1.187720e-14                            -999.0   
4                          2.418840e-14                            -999.0   

       nufnu_1.5keV(erg.s-1.cm-2)  nufnu_error_1.5keV(erg.s-1.cm-2)  \
OBJID                                                                 
0                    1.088260e-13                      4.027270e-14   
1                    7.075490e-14                      3.430380e-14   
2                    4.655630e-14                      3.005940e-14   
3                    9.568910e-14                      3.936350e-14   
4                    4.673000e-14                      2.649940e-14   

       upper_limit_1.5keV(erg.s-1.cm-2)  nufnu_4.5keV(erg.s-1.cm-2)  \
OBJID                                                                 
0                                -999.0                9.370070e-14   
1                                -999.0                1.066430e-13   
2                                -999.0                9.353370e-14   
3                                -999.0                1.442380e-13   
4                                -999.0                9.390950e-14   

       nufnu_error_4.5keV(erg.s-1.cm-2)  upper_limit_4.5keV(erg.s-1.cm-2)  
OBJID                                                                      
0                          4.601960e-14                            -999.0  
1                          5.170340e-14                            -999.0  
2                          5.227960e-14                            -999.0  
3                          5.933510e-14                            -999.0  
4                          4.601620e-14                            -999.0

In [36]:
df_sds_matched = df_sds.loc[objids_gc.index]
df_sds_matched.head()

RA       DEC            NH  ENERGY_SLOPE ENERGY_SLOPE_ERROR  \
OBJID_A                                                                       
14       -3.756921 -0.026718  3.320000e+20           0.8          -999/-999   
45       24.693637 -0.672514  2.650000e+20           0.8          -999/-999   
46       24.657625 -0.668684  2.650000e+20           0.8          -999/-999   
47       24.819667 -0.621067  2.670000e+20           0.8          -999/-999   
48       24.696104 -0.495326  2.620000e+20           0.8          -999/-999   

         EXPOSURE_TIME  nufnu_3keV(erg.s-1.cm-2)  \
OBJID_A                                            
14              4826.6              5.258800e-14   
45             17176.0              9.711220e-15   
46             17138.0              1.439760e-14   
47             15367.0              1.321040e-14   
48             15398.6              2.047130e-14   

         nufnu_error_3keV(erg.s-1.cm-2)  nufnu_0.5keV(erg.s-1.cm-2)  \
OBJID_A                                                               
14                         1.401600e-14                7.640390e-14   
45                         3.698730e-15                4.282920e-15   
46                         4.383680e-15                4.616470e-15   
47                         4.249480e-15                9.999910e-15   
48                         5.336760e-15                1.202760e-14   

         nufnu_error_0.5keV(erg.s-1.cm-2)  upper_limit_0.5keV(erg.s-1.cm-2)  \
OBJID_A                                                                       
14                           2.363380e-14                            -999.0   
45                           3.262950e-15                            -999.0   
46                           3.287120e-15                            -999.0   
47                           5.003590e-15                            -999.0   
48                           5.394080e-15                            -999.0   

         nufnu_1.5keV(erg.s-1.cm-2)  nufnu_error_1.5keV(erg.s-1.cm-2)  \
OBJID_A                                                                 
14                     2.679660e-14                      1.657780e-14   
45                     9.418440e-15                      5.908220e-15   
46                     1.691710e-14                      7.650940e-15   
47                     1.951080e-14                      8.467530e-15   
48                     1.766150e-14                      7.920770e-15   

         upper_limit_1.5keV(erg.s-1.cm-2)  nufnu_4.5keV(erg.s-1.cm-2)  \
OBJID_A                                                                 
14                                 -999.0                8.094140e-14   
45                                 -999.0                1.432350e-14   
46                                 -999.0                2.058300e-14   
47                                 -999.0                1.887070e-14   
48                                 -999.0                2.686790e-14   

         nufnu_error_4.5keV(erg.s-1.cm-2)  upper_limit_4.5keV(erg.s-1.cm-2)  
OBJID_A                                                                      
14                          -5.387110e-11                      2.677380e-13  
45                           8.985180e-15                     -9.990000e+02  
46                           1.034270e-14                     -9.990000e+02  
47                          -1.668350e-11                      4.911330e-14  
48                           1.204960e-14                     -9.990000e+02

In [37]:
df_sds_nonmatched = df_sds.loc[~df_sds.index.isin(objids_gc.index)]
df_sds_nonmatched.head()

RA       DEC            NH  ENERGY_SLOPE ENERGY_SLOPE_ERROR  \
OBJID                                                                       
0      14.102000 -1.277310  3.540000e+20           0.8          -999/-999   
1      14.079733 -1.249499  3.520000e+20           0.8          -999/-999   
2      14.095850 -1.227643  3.510000e+20           0.8          -999/-999   
3      14.068408 -1.234838  3.510000e+20           0.8          -999/-999   
4      14.070508 -1.221223  3.500000e+20           0.8          -999/-999   

       EXPOSURE_TIME  nufnu_3keV(erg.s-1.cm-2)  \
OBJID                                            
0             4572.8              7.969610e-14   
1             4684.2              6.277090e-14   
2             4649.6              5.501970e-14   
3             4642.0              8.173270e-14   
4             4587.3              7.365840e-14   

       nufnu_error_3keV(erg.s-1.cm-2)  nufnu_0.5keV(erg.s-1.cm-2)  \
OBJID                                                               
0                        2.117700e-14                2.247330e-14   
1                        1.975260e-14                1.277270e-14   
2                        1.974560e-14                2.238590e-14   
3                        2.256640e-14                1.151060e-14   
4                        2.115000e-14                5.616880e-14   

       nufnu_error_0.5keV(erg.s-1.cm-2)  upper_limit_0.5keV(erg.s-1.cm-2)  \
OBJID                                                                       
0                          1.552740e-14                            -999.0   
1                          1.252210e-14                            -999.0   
2                          1.751890e-14                            -999.0   
3                          1.187720e-14                            -999.0   
4                          2.418840e-14                            -999.0   

       nufnu_1.5keV(erg.s-1.cm-2)  nufnu_error_1.5keV(erg.s-1.cm-2)  \
OBJID                                                                 
0                    1.088260e-13                      4.027270e-14   
1                    7.075490e-14                      3.430380e-14   
2                    4.655630e-14                      3.005940e-14   
3                    9.568910e-14                      3.936350e-14   
4                    4.673000e-14                      2.649940e-14   

       upper_limit_1.5keV(erg.s-1.cm-2)  nufnu_4.5keV(erg.s-1.cm-2)  \
OBJID                                                                 
0                                -999.0                9.370070e-14   
1                                -999.0                1.066430e-13   
2                                -999.0                9.353370e-14   
3                                -999.0                1.442380e-13   
4                                -999.0                9.390950e-14   

       nufnu_error_4.5keV(erg.s-1.cm-2)  upper_limit_4.5keV(erg.s-1.cm-2)  
OBJID                                                                      
0                          4.601960e-14                            -999.0  
1                          5.170340e-14                            -999.0  
2                          5.227960e-14                            -999.0  
3                          5.933510e-14                            -999.0  
4                          4.601620e-14                            -999.0

In [38]:
print(len(df_sds_matched))
print(len(df_sds_nonmatched))

1105
1659


In [39]:
from bokeh.io import show
from numpy import log10 as log

_col = 'nufnu_3keV(erg.s-1.cm-2)'

data1 = df_sds_nonmatched[_col].dropna()
data2 = df_sds_matched[_col].dropna()
                          
legend1 = 'non-matched ({:d})'.format(len(data1))
legend2 = 'matched ({:d})'.format(len(data2))

p = plot_distros(log(data1), log(data2), legend1, legend2)

_cnm = _col.split('(')[0]
p.title.text = 'SDS x-ray ({}) flux distribution'.format(_cnm)
p.title.align = 'center'
p.xaxis.axis_label = 'log( nuFnu_3keV [erg.s-1.cm-2] )'
p.yaxis.axis_label = '#'

p.legend.location = 'top_right'

show(p)

In [40]:
from bokeh.io import show
from numpy import log10 as log

_col = 'NH'

data1 = df_sds_nonmatched[_col].dropna()
data2 = df_sds_matched[_col].dropna()
                          
legend1 = 'non-matched ({:d})'.format(len(data1))
legend2 = 'matched ({:d})'.format(len(data2))

p = plot_distros(data1, data2, legend1, legend2)

p.title.text = 'SDS NH distribution'
p.title.align = 'center'
p.xaxis.axis_label = '{}'.format(_col)
p.yaxis.axis_label = '#'

p.legend.location = 'top_right'

show(p)

In [41]:
from bokeh.io import show
from numpy import log10 as log

_col = 'EXPOSURE_TIME'

data1 = df_sds_nonmatched[_col].dropna()
data2 = df_sds_matched[_col].dropna()
                          
legend1 = 'non-matched ({:d})'.format(len(data1))
legend2 = 'matched ({:d})'.format(len(data2))

p = plot_distros(data1, data2, legend1, legend2)

p.title.text = 'SDS exposure-time distribution'
p.title.align = 'center'
p.xaxis.axis_label = '{}'.format(_col)
p.yaxis.axis_label = '#'

p.legend.location = 'top_right'

show(p)

In [42]:
from bokeh.io import show
from numpy import log10 as log

_col = 'nufnu_0.5keV(erg.s-1.cm-2)'

data1 = df_sds_nonmatched[_col].dropna()
data1 = data1.loc[data1>0]
data2 = df_sds_matched[_col].dropna()
data2 = data2.loc[data2>0]
                          
legend1 = 'non-matched ({:d})'.format(len(data1))
legend2 = 'matched ({:d})'.format(len(data2))

p = plot_distros(log(data1), log(data2), legend1, legend2)

_cnm = _col.split('(')[0]
p.title.text = 'SDS x-ray ({}) flux distribution'.format(_cnm)
p.title.align = 'center'
p.xaxis.axis_label = 'log( {} )'.format(_col)
p.yaxis.axis_label = '#'

p.legend.location = 'top_right'

show(p)

In [43]:
from bokeh.io import show
from numpy import log10 as log

_col = 'nufnu_1.5keV(erg.s-1.cm-2)'

data1 = df_sds_nonmatched[_col].dropna()
data1 = data1.loc[data1>0]
data2 = df_sds_matched[_col].dropna()
data2 = data2.loc[data2>0]
                          
legend1 = 'non-matched ({:d})'.format(len(data1))
legend2 = 'matched ({:d})'.format(len(data2))

p = plot_distros(log(data1), log(data2), legend1, legend2)

_cnm = _col.split('(')[0]
p.title.text = 'SDS x-ray ({}) flux distribution'.format(_cnm)
p.title.align = 'center'
p.xaxis.axis_label = 'log( {} )'.format(_col)
p.yaxis.axis_label = '#'

p.legend.location = 'top_right'

show(p)

In [44]:
def plot_distro_flux(title, data1, data2, colname, legend1, legend2):
    from numpy import log10 as log

    _col = colname
    data1 = data1.loc[data1>0]
    data2 = data2.loc[data2>0]

    p = plot_distros(log(data1), log(data2), legend1, legend2)

    p.title.text = title
    p.title.align = 'center'
    p.xaxis.axis_label = 'log( {} )'.format(_col)
    p.yaxis.axis_label = '#'

    p.legend.location = 'top_right'

    show(p)

_col = 'nufnu_4.5keV(erg.s-1.cm-2)'
_cnm = _col.split('(')[0]
title = 'SDS x-ray ({}) flux distribution'.format(_cnm)

data1 = df_sds_nonmatched[_col].dropna()
data2 = df_sds_matched[_col].dropna()
legend1 = 'non-matched ({:d})'.format(len(data1))
legend2 = 'matched ({:d})'.format(len(data2))

from bokeh.io import show
p = plot_distro_flux(title, data1, data2, _col, legend1, legend2)